lib/parser.js

```diff
@@ -299,19 +301,42 @@ Parser.prototype._trimNodes = function (node) {
   }
 };
 
+Parser.prototype._initializeUserVariables = function(filePath, config) {
+  const userDefinedVariables = {};
+  userDefinedVariables['baseUrl'] = '{{baseUrl}}';
```

Comment:
1. Does this correctly solve the `baseUrl` problem?
2. The next 2 lines are about `baseUrl` for a different purpose — is that why it's named `base`?
   Let's place this line (with a comment) just before `return userDefinedVariables;` instead.

Reply:
1. Yes. I tested printing {{ baseUrl }} in the site with and without this line.
2. The next 2 lines are for using the correct variables.md if there is a subsite. I will move the first line down